### PR TITLE
🌱 Move make target

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Moves the generate-e2e-templates target into the generate section of the makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1148,12 +1151,11 @@ $(GINKGO): # Build ginkgo from tools folder.
 ## --------------------------------------
 DOCKER_TEMPLATES := test/e2e/data/infrastructure-docker
 
-.PHONY: generate-e2e-templates
-generate-e2e-templates: $(KUSTOMIZE) generate-e2e-templates-v1alpha3 generate-e2e-templates-v1alpha4 generate-e2e-templates-v1.2 generate-e2e-templates-v1beta1 ## Generate cluster templates for all versions
-
+.PHONY: generate-e2e-templates-v1alpha3
```
Comment (sbueringer) [inline Makefile:1154]: Should we move all generate targets in the generate section? Seems inconsistent to me to just move one of them

File: Makefile [hunk 2]
```diff
@@ -463,6 +463,9 @@ generate-modules: ## Run go mod tidy to ensure modules are up to date
 	cd $(TOOLS_DIR); go mod tidy
 	cd $(TEST_DIR); go mod tidy
 
+.PHONY: generate-e2e-templates
+generate-e2e-templates: $(KUSTOMIZE) generate-e2e-templates-v1alpha3 generate-e2e-templates-v1alpha4 generate-e2e-templates-v1.2 generate-e2e-templates-v1beta1 ## Generate cluster templates for all versions
```
Comment (oscr) [inline Makefile:467]: Suggestion to use addprefix function:
`generate-e2e-templates: $(KUSTOMIZE) $(addprefix generate-e2e-templates-, v1alpha3 v1alpha4 v1.2 v1beta1)`

Expansion can be verified `make --debug generate-e2e-templates`

File: Makefile (hunk 1 above)
Comment (fabriziopandini) [reply Makefile:1154]: see https://github.com/kubernetes-sigs/cluster-api/pull/7329#discussion_r986835619
fine with both approaches for me

Comment (sbueringer) [reply Makefile:1154]: I would prefer keeping them in one block (and in generate it makes more sense)

Comment (fabriziopandini) [reply Makefile:1154]: kk , sooner or later we should break down our make file in smaller mk files and import them

Comment (oscr) [reply Makefile:1154]: I like what the OpenStack provider has done with it's Makefiles. 

For example:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/common.mk
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/versions.mk

I would be willing to make a draft for a similar change once all the other Makefile changes are merged.

**Review by oscr — APPROVED** [review]
Looks great!

/lgtm

**Pull request merged**